### PR TITLE
Feature/chart data

### PR DESCRIPTION
**Pull request by allenan**
resolves https://github.com/helium/hotspot-app/issues/88

depends on wallet api PR: https://github.com/helium/wallet-api/pull/13

### CODE REVIEW

File: src/components/BarChart/WalletChart.tsx [hunk 1]
```diff
@@ -1,34 +1,53 @@
...
+  }, [dispatch, timeframe, filter])
 
   const headerHeight = 30
   const padding = 20
-  const chartHeight = height - headerHeight - padding
+  const chartHeight = useMemo(() => height - headerHeight - padding, [height])
 
-  const changeTimeframe = (t: number) => {
-    setTimeframe(t)
+  const changeTimeframe = useCallback((range: ChartRange) => {
```
Comment (matthewcarlreetz) [inline src/components/BarChart/WalletChart.tsx:44]: you could curry this function so you don't have to have inline functions
```
  const changeTimeframe = useCallback(
    (range: ChartRange) => () => {
      setTimeframe(range)
      triggerImpact()
    },
    [],
  )
```

Then just call it right away...
```
 <TouchableWithoutFeedback onPress={changeTimeframe('daily')}>
 ```

File: src/components/BarChart/WalletChart.tsx [hunk 2]
```diff
@@ -66,60 +85,47 @@ const WalletChart = ({ height }: Props) => {
               >
                 <CarotRight width={12} height={12} style={{ marginRight: 4 }} />
```
Comment (matthewcarlreetz) [inline src/components/BarChart/WalletChart.tsx:86]: Would be a good idea to get rid of some of the inline styling

```
              <Box flexDirection="row" alignItems="center">
                <CarotRight width={12} height={12} />
                <Text variant="body2" marginLeft="s">
                  {round(focusedData?.down, 2).toLocaleString()}
                </Text>
              </Box
```

File: src/components/BarChart/WalletChart.tsx [hunk 3]
```diff
@@ -66,60 +85,47 @@ const WalletChart = ({ height }: Props) => {
...
         </Box>
       </Box>
+      {data.length === 0 && (
+        <Box
+          height={chartHeight}
+          width="100%"
+          justifyContent="center"
+          position="absolute"
+        >
+          <ActivityIndicator />
```
Comment (matthewcarlreetz) [inline src/components/BarChart/WalletChart.tsx:119]: ActivityIndicator needs a color prop in order to show on Android

File: src/components/BarChart/WalletChart.tsx [hunk 4]
```diff
@@ -1,34 +1,53 @@
...
-    setTimeframe(t)
+  const changeTimeframe = useCallback((range: ChartRange) => {
+    setTimeframe(range)
     triggerImpact()
-  }
+  }, [])
 
-  const handleFocusData = (data: ChartData | null): void => {
-    setFocusedData(data)
+  const handleFocusData = (chartData: ChartData | null): void => {
```
Comment (matthewcarlreetz) [inline src/components/BarChart/WalletChart.tsx:49]: Could `useCallback` here

File: src/components/BarChart/WalletChart.tsx [hunk 5]
```diff
@@ -66,60 +85,47 @@ const WalletChart = ({ height }: Props) => {
...
-  })),
-  2: times(12).map((v, i) => ({
-    up: random(0, 100),
-    down: random(0, 1) ? random(0, 40) : 0,
-    day: weekdays[i % 7],
-    id: [2, i].join('-'),
-  })),
-}
-
 export default WalletChart
```
Comment (matthewcarlreetz) [inline src/components/BarChart/WalletChart.tsx:131]: might want to memoize this?

File: src/utils/walletClient.ts [hunk 6]
```diff
@@ -12,6 +13,7 @@ const makeRequest = async (url: string, opts: RequestInit) => {
     }
 
     const route = [Config.WALLET_API_BASE_URL, url].join('/')
+    // const route = ['http://localhost:4000/api', url].join('/')
```
Comment (matthewcarlreetz) [inline src/utils/walletClient.ts:16]: remove?

**Review by matthewcarlreetz — APPROVED** [review]
Just a few suggestions for you. Everything seems to be working well.

There may be one alignment issue...
iPhone 11 2021-02-17 at 2 22 30 PM

**Comment by allenan** [comment]
> Just a few suggestions for you. Everything seems to be working well.
> 
> There may be one alignment issue...
> iPhone 11 2021-02-17 at 2 22 30 PM

damn ok I'll tilt at the alignment windmill some more. mine are the reverse, with the blue bars being the largest and the green dots at the top.

**Pull request merged**